pyouroboros/dockerclient.py

```diff
@@ -237,6 +237,41 @@ def update(self):
 
                 created = self.client.api.create_container(**new_config)
                 new_container = self.client.containers.get(created.get("Id"))
+
+                # disconnect new container from old networks (with possible
+                # broken config)
+                for network in new_container.attrs['NetworkSettings']['Networks']:
+                    self.client.api.disconnect_container_from_network(
+                        container=new_container.attrs['Id'],
+                        net_id=network,
+                        force=True
+                    )
+
+                # connect the new container to all networks of the old container
+                for network in container.attrs['NetworkSettings']['Networks']:
+                    host_net = self.client.networks.get(network_id=network)
+                    if (host_net.attrs.get('Options')
+                        and 'true' == host_net.attrs['Options'].get('com.docker.network.bridge.default_bridge')
```

Comment:
almost there! wrapping the 2 separate operators isnt needed with () if thing_one and thing_2 == ‘true’ will achieve the same results. wrapping is generally needed with or more often.

Reply:
The parentheses are only there because I wrapped the if-statement according to [PEP8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length), not in order to group the boolean logic.  
Is it ok for you to keep them after this explanation? :wink:

Reply:
you talking about 79 char line length? that is very very old hat and the standard is now 120 chars. even if you stuck with the 79 the wrapping wouldnt be needed

Reply:
> you talking about 79 char line length?

Yep.



> even if you stuck with the 79 the wrapping wouldnt be needed

Sure?
```shell
› echo "                    if host_net.attrs.get('Options') and 'true' == host_net.attrs['Options'].get('com.docker.network.bridge.default_bridge'):"|wc -m
142
```

Don't get me wrong, I will change the code however you want it to be. I'm just arguing :wink:

Reply:
See also these visual indicators in my editor:  
![screenshot_20190209_235548](https://user-images.githubusercontent.com/4653434/52527283-3a040700-2cc6-11e9-8a79-621d7067b254.png)

Reply:
I jumped on the computer to reply this time so i could show you. Python is wonderful in many ways. One of them is nested lookups. You can rewrite that entire line to show like this:
```py
if host_net.attrs.get('Options', {}).get('com.docker.network.bridge.default_bridge') == 'true':
```
This not only achieves the initial check of options, but also gives an empty dict to allow the nested lookup of the default_bridge which will show false but work as expected. ;)

Reply:
Well, that's definitely more elegant than my approach.  
I changed the if-statement accordingly. :slightly_smiling_face: